sdk/src/Handlers/AwsSdk/Internal/XRayPipelineHandler.cs

```diff
@@ -394,6 +395,13 @@ private void ProcessEndRequest(IExecutionContext executionContext)
             else
             {
                 subsegment.Aws["request_id"] = responseContext.Response.ResponseMetadata.RequestId;
+
+                // try getting x-amz-id-2 if dealing with s3 request
+                if (responseContext.Response.ResponseMetadata.Metadata.TryGetValue(S3ExtendedRequestIdHeaderKey, out string extendedRequestId))
+                {
+                    subsegment.Aws[ExtendedRquestIdSegmentKey] = extendedRequestId;
```

Comment:
I'm slightly confused here. When we try and get request ids from the request headers, the S3 request headers will have "x-amz-request-id" instead of the "x-amzn-RequestId" and the additional "x-amz-id-2". Whereas when there is a response, the response metadata will have the request id in "responseContext.Response.ResponseMetadata.RequestId" attribute irrespective of it being S3 request or not, and the additional metadata with "x-amz-id-2" key in case of S3.
Is it correct?

Reply:
> I'm slightly confused here. When we try and get request ids from the request headers, the S3 request headers will have "x-amz-request-id" instead of the "x-amzn-RequestId" and the additional "x-amz-id-2". Whereas when there is a response, the response metadata will have the request id in "responseContext.Response.ResponseMetadata.RequestId" attribute irrespective of it being S3 request or not, and the additional metadata with "x-amz-id-2" key in case of S3.
> Is it correct?

That's correct

Reply:
> We have a constant for "id_2" but not for "request_id" although the latter is used in multiple places. It would be good to make the consistent. Either we have constants for both or for none. I would prefer having a constant for any string that is being referenced in multiple places unchanged.

I would prefer non constant for all key strings as how we handler other key strings, like `region`, `operation`, etc.

Reply:
Well yeah, ideally we should have had a constant class with all the keys of the attributes we record. Very much like the [SemanticConventions](https://github.com/open-telemetry/opentelemetry-dotnet/blob/d24a0429642f65b1e107df699ddfc3c5cf898b62/src/OpenTelemetry.Api/Internal/SemanticConventions.cs) class in OpenTelemetry. But this is okay for now since we do not touch these attributes very often.